### DISCUSSION TITLE
Convert Discussion into Issue

### DISCUSSION THREAD

**Discussion by agentpoyo** [community/community#2861] [Discussions] [answered]
Seems like one can convert an issue into a discussion but not vice versa. This would be a great feature I think, particularly if community members are providing ideas that could turn into Issues (Action Items).

**Comment by evi-liu** [comment]
Thanks for the suggestion! What context would you like to bring from Discussions into a converted issue?

**Reply by agentpoyo** [reply]
Perhaps an idea from the original topic starter would be used is what I'd imagine. Wouldn't need followup comments unless a comment has context within that would make it into the issue. It would be nice if you convert a discussion, you could select what comments you want to  be transferred over so it's not cluttered (if there are a lot of comments).

**Reply by evi-liu** [reply]
Do you have any example Discussions that you wish you could create an issue from?

**Reply by gjaekel** [reply]
@evi-liu I may give one (also): https://github.com/igniterealtime/openfire-pade-plugin/discussions/234 . At this project we "opened" the discussion section a few days before. There a just a few entries yet, but at once there is such a candidate.

In my eyes this is a typical one for the use case where a vanilla user choose the wrong vehicle - the developer wish to handle most parts of this as an issue, because it's an individual problem. But there's also some meta discussion, that might be a candidate to leave as an discussion. But for the first sprint, a "full conversion" between the sections will be a big step forward and fulfill most needs.

**Reply by michael-hawker** [reply]
Here's another example of a submitted suggestion/idea, that we now then want to go work on and convert into an issue: https://github.com/windows-toolkit/WindowsCommunityToolkit/discussions/3928

**Comment by WrichikBasu** [comment]
I second this suggestion. Some reasons why it might be necessary to port a discussion to an issue:

1. Someone mistakenly posts an issue (e.g. bug report) as a discussion.
1. Someone asks a question, and eventually it turns out that there is a bug. It would be great if I can also hide comments that are out of context/not relevant.

**Reply by duncanmcclean** [reply]
Yeah, I've had a couple of people who report bugs as part of Discussions, where I'd prefer them to be organised as issues instead.

**Reply by evi-liu** [reply]
Would you prefer if the discussions was still kept there? Or would you want the discussion to be deleted once the issue has been opened? 

Additionally, what context would you want to bring from the discussion to the issue?

**Reply by WrichikBasu** [reply]
@evi-liu Definitely delete the discussion once it has been converted to an issue. URLs that used to point to the discussion should now point to the issue instead. The following would be the best: transfer all comments from discussion to issue, but provide an option to the repo maintainer to hide certain comments. In this way, history will be preserved, but out-of-context stuff can be hidden.

**Reply by behdad** [reply]
> Would you prefer if the discussions was still kept there? Or would you want the discussion to be deleted once the issue has been opened?

My use-case is like @WrichikBasu.

> Additionally, what context would you want to bring from the discussion to the issue?

Basically, member of community opens a Discussion item to ask why something doesn't work. We developers realize it's a bug and want to fix it. The original Discussion has all the data and discussion that belongs to an issue.

**Reply by bagder** [reply]
Some users are also very careful and want to be gentle. They rather start a discussion to ask about it rather than to downright claim it is an issue at start, while members of the project might recognize or consider it as an issue and therefore it should be treated as such and moved. Recent example in the curl project: https://github.com/curl/curl/discussions/6468

**Reply by Marcono1234** [reply]
> Definitely delete the discussion once it has been converted to an issue.

But then it would really be necessary to copy over all comments (maybe with the optional feature to hide some of them, like you proposed).
Imagine this scenario: A user creates a discussion because they encountered an issue. You guide them through debugging steps and provide useful advice. It turns out they encountered a bug and you convert the discussion to an issue.
Then it would still be valuable to the user to have access to the debugging steps and advice you gave them in the discussion (e.g. when they are applicable to other scenarios as well), even though they are not relevant for the newly created issue.

**Reply by WrichikBasu** [reply]
@Marcono1234 Yes, all the comments should be copied into the issue.

**Reply by James-E-A** [reply]
I have a clear-cut example of this use-case: I posted a Discussion, then later discover it **should have been filed as an Issue in the first place.**

https://www.github.com/dependabot/dependabot-core/discussions/9345#discussioncomment-9214853

**Comment by asmeurer** [comment]
It's very common for users, especially new users to a software, to not realize that their "question" is actually a bug report because the behavior or error message they got isn't actually supposed to happen. Even experienced developers can be confused about what is or isn't intended behavior is [a bug or a feature](https://en.wikipedia.org/wiki/Undocumented_feature) if they aren't experts in the code in question. 

This can also happen in the other direction: a developer may incorrectly think an issue is a question and convert it to a discussion, only to later realize it is actually a bug.

**Reply by evi-liu** [reply]
Would you prefer if the discussions was still kept there? Or would you want the discussion to be deleted once the issue has been opened?

Additionally, what context would you want to bring from the discussion to the issue?

**Reply by tech2serve** [reply]
How about provide the option to transfer the topic completely, or split the two? 

This may be better for a broader topic that results in multiple issues being generated.

**Comment by xetra11** [comment]
Need !!

**Comment by joshbruce** [comment]
See also: https://github.com/github/feedback/discussions/2988

Found another seeming to ask for similar but that was before I learned cross-linking doesn’t work and this one appears to have more activity.

**Comment by philsward** [comment]
One thought is to have a "create issue" button and that new issue automatically creates a link to the discussion.

Part of me wants the full discussion in the issue, however the other part of me realizes that it may be way too much info to weed through and the issue itself might be a culmination of the thoughts within the discussion so having a fresh starting point might make the most sense.

@agentpoyo has a good idea worth considering.

**Reply by evi-liu** [reply]
What about if you were able to highlight select pieces of context from the discussion to include into the issue?

**Reply by philsward** [reply]
@evi-liu I think as long as it doesn't take a long time, it would be a good approach. Easy for short discussions, time consuming for long, so the "option" to choose context would be a good.

**Reply by xetra11** [reply]
That's a cool idea! Can't be that complex as a feature I'd assume!

**Comment by behdad** [comment]
We also want this. Example: https://github.com/harfbuzz/harfbuzz/discussions/2807

User opened discussion asking why something doesn't work. We want to fix it, so want to turn into issue.

**Comment by jay** [comment]
I would like to see this as well. Example:

https://github.com/curl/curl/discussions/6468

/cc @bagder

**Comment by wonderflow** [comment]
Hi, @evi-liu I'm coming here to the github feedback forum the first time just for this feature. It's very userful. 

In my use case, the new coming users didn't realize their queston is a feature request or a bug. If we have decide to implement it, that's not in the scope of discussion anymore, it's just an issue.

I don't think we need to keep it in discussion if it has been coverted to an issue.

**Comment by michael-hawker** [comment]
We're thinking of using Discussions as an entry point for topics on our repo, and then leave Issues for specifically bugs or work that we've planned out into milestones/projects.

So if an idea that starts in Discussions becomes prevalent enough, has support, or a community member picks it up to work on, it'd be great to convert from a Discussion into an Issue and maintain that initial conversation as part of the history of the work.

**Reply by StjerneIdioten** [reply]
Exactly this. I am working on an addon for Blender, where I encourage people to post anything that might make their life easier in using it. I was hoping to use the discussion forum as a stepping stone for feature requests. So I could decide on what to work with next, depending on what people show the most interest in.

**Reply by jsmorabito** [reply]
echoing this. we use discussions as a staging area for bugs and feature requests to guage interest/support and to prevent our issues section from being overwhelmed since it is a place of action for our project. for reference i am coming from athens research which is an open source project https://github.com/athensresearch

**Comment by gjaekel** [comment]
>     1. Someone mistakenly posts an issue (e.g. bug report) as a discussion.

This is **exactly** the situation I faces since 30min ... :)

**Comment by fabratu** [comment]
This would be a great feature indeed. We are now using discussions on our project and especially the incorrectly created/converted discussions is a very realistic use case.

**Comment by PranamBhat** [comment]
There are 2 reasons why it might be necessary to convert a Discussion to an Issue:

1) Someone mistakenly posts an issue (e.g. bug report) as a discussion.
2) Someone asks a question, and eventually it turns out that there is a bug. It would be great if I can also hide comments that are out of context/not relevant.

This will be a very rare scenario. If this in case happens, We have a "**Delete**" option. So, delete the **Discussion/Issue** and post them once again accordingly.

**Reply by bronger** [reply]
Indeed, it is not a solution.  I recently moved a discussion thread to the issues manually, marking who said what. This is sub-optimal, to say the least.

**Comment by lindeloev** [comment]
I second this suggestion. Example: https://github.com/lindeloev/mcp/discussions/109. The user asked about a solution to a problem. After a quick back-and-forth, it turns out that I would need to develop a new feature.

**Reply by rarkins** [reply]
You can. Ask support to enable it

**Reply by jsmorabito** [reply]
Can you elaborate? Is the ability to convert discussions back into issues already available?

**Reply by rarkins** [reply]
No, I was responding to someone who said issue -> discussion (not vice versa). It used to be available, then was removed due to a bug. I believe you can ask support to enable it for your org or repo specifically, unless it's back by default. 

As far as I know we're all still waiting for discussion -> issue

**Reply by jsmorabito** [reply]
ah got it! yea i can do issue -> discussion. still waiting for reverse then!

**Comment by emiltin** [comment]
Yes. I'm reluctant to try Discussions and convert existing issues to discussions if I know it's a one-way operation.

**Comment by nrdxp** [comment]
We need a completely reversible operation or this isn't always great for what it's meant for since someone, at some time, will inevitably turn something that is better suited as an issue into a discussion, and the inability to turn it back will make anyone reluctant of moving to a discussion in the first place _even though_  the distinction may better suit it.

A simple proposed implementation could be to simply lock the discussion and reopen the issue if it was previously converted into a discussion, while updating the issue with the discussions thread additions on reopen.

The current banner above converted issues seems sufficient to alert users that the discussion is locked and the issue reopened, in the same way it communicates the inverse now.

**Comment by erwincoumans** [comment]
I just converted all issues into discussions, by batch, assuming it was reversible. Hope it will be implemented.

**Comment by adityaruplaha** [comment]
I too support this, along with some way to "archive" a discussion so that it doesn't show up in View All.

**Comment by helmasaur** [comment]
Same here! First of all, it's interesting if someone makes a mistake. It quite confusing at the first sight to understand the different between the "Issue" and "Discussion" for someone familiar with GitHub, so imagine someone who just wants to give a feedback but who is not a developer.

Also, it would be interesting if an idea is validated, to convert it to an issue. I am not even sure it is possible to link a discussion to an issue or a commit.

**Comment by evi-liu** [comment, marked as answer]
### `Create issue from discussion` Private Beta Available! 🚀 

Hi folks, thanks for all of the helpful feedback on this! I'm happy to announce that we have a private beta available to enable creating an issue from a discussion. With this feature, you'll see a new button on your sidebar for `Create issue from Discussion`. See image below for what the experience will look like.  Please note that this is just the first iteration. With your feedback, we'd love to know what else would be valuable in this experience. If you'd like to help us private beta test, please comment below with the repository you'd like to test on. Private beta is limited to the first 15 responders for now. 

Click new `Create issue from discussion` button in the Discussions sidebar:
![image](https://user-images.githubusercontent.com/8809849/117064279-d321fc80-acda-11eb-95b9-7d6df92227a9.png)

Takes you to an Issue draft with the linked Discussion and context of the post: 
![image](https://user-images.githubusercontent.com/8809849/117078673-2a31cc80-acef-11eb-8a8c-85a93525d1bf.png)

New issue created:
![image](https://user-images.githubusercontent.com/8809849/117079022-d1aeff00-acef-11eb-97d2-c3fe1e49e979.png)

**Reply by michael-hawker** [reply]
🎉 Thanks @evi-liu! This is going to be great. If we're already using Discussions should this appear now, or is there a separate step we need to take to enable it?

Also, if a Discussion was previously created from an existing Issue, does it re-open the old issue or create a new issue?

**Reply by helmasaur** [reply]
Wow, great news! And I am amazed how fast it has been done! Thank you very much.

I agree to participate to the beta and test the new functionnaly and give my feedback.

**Reply by rarkins** [reply]
I'd like to test this out in github.com/renovatebot/renovate - thanks

**Reply by adityaruplaha** [reply]
I'd like to have this on https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher, please.

**Reply by gjaekel** [reply]
I'd like to evaluate for https://github.com/igniterealtime/openfire-pade-plugin

**Reply by emiltin** [reply]
Thanks for working on this. Though part of me is wondering whether it would be simpler and easier to have only the issue list, but be able to mark items as discussions, and filter by issue/discussion. Then there woukd be no need for this kind of transfer between issue/discussion.

**Reply by jsmorabito** [reply]
would like to test this out with https://github.com/athensresearch/athens thanks so much!

**Reply by evi-liu** [reply]
✅ all of the repos requested above have been flagged in, so you should have access to this feature now!

@michael-hawker your repository will have to be flagged in. could you post the repository you'd like to test this on? 

@helmasaur which repository would you like to enable this on? 

@emiltin for now, we'd like to keep the spaces separate as to not clog the issues view, but this is great feedback to noodle on.

**Reply by agentpoyo** [reply]
Thanks @evi-liu , any way to add https://github.com/cncf/cnf-testsuite as well, since I opened this original discussion/issue?  :+1:

**Reply by drakkan** [reply]
Hi, would like to test this out with [https://github.com/drakkan/sftpgo](https://github.com/drakkan/sftpgo/) thanks!

**Reply by evi-liu** [reply]
@agentpoyo @drakkan both repos have been flagged in!

**Reply by gjaekel** [reply]
@evi-liu Thank you for "flagging" me!

OK: As the title annouces, this will **Create an Issue** with the first entry of the Discussion and a link to it. But this don't **convert** the Discussion into an Issue as requested here IMHO, it will just produce another, new object.

I'm curious about your next iteration.

**Reply by michael-hawker** [reply]
@evi-liu thanks! https://github.com/windows-toolkit/WindowsCommunityToolkit is our repo. Thanks! FYI @kyaa-dost

**Reply by helmasaur** [reply]
@evi-liu Sorry, I didn't give the repo. So it's https://github.com/helmasaur/horsengel-roulette.

**Reply by wonderflow** [reply]
https://github.com/oam-dev/kubevela Thanks! @evi-liu

**Reply by Kyaa-dost** [reply]
Hi @evi-liu! Still waiting on this one https://github.com/windows-toolkit/WindowsCommunityToolkit
Please let us know when it's ready 🚀

**Reply by evi-liu** [reply]
@michael-hawker @helmasaur @wonderflow @Kyaa-dost ✅ your repos have all been flagged in now, so you should see the button available. 

> I'm curious about your next iteration.

@gjaekel yes this is our first iteration! based on usage and customer feedback, we very well may add the ability to fully convert a discussion into an issue :)

**Reply by Kyaa-dost** [reply]
@evi-liu Thanks! I can confirm that this is accessible now 🚀

**Reply by igoreliasm** [reply]
💯

**Reply by joshbruce** [reply]
Hello - could we add it here??

https://github.com/markedjs/marked

**Reply by katelynnbarlowe** [reply]
Looks like you've reached your limit, but if not:
https://github.com/deliciousbrains/spinupwp

We're using discussions to manage app feature requests, so this would be perfect for us. And I'd love to give feedback on your app changes too! 😄

**Reply by broofa** [reply]
Please enable this for https://github.com/broofa/flightcard !  🙏

**Reply by oseibonsu** [reply]
@evi-liu Please enable for https://github.com/Acadeum/Tickets  Thanks!

**Reply by JRoy** [reply]
Hey, could you enable this for [EssentialsX/Essentials](https://github.com/EssentialsX/Essentials) :)

**Reply by heitorlessa** [reply]
I'd love to be able to use this too as I'm in the process of transferring all discussions as issues to another repo: https://github.com/awslabs/aws-lambda-powertools-python, pretty please :)

**Reply by evi-liu** [reply]
thanks for all of the interest! all of the repos in this thread have been flagged in :)

**Reply by kbuzard** [reply]
If this is still open, I'd love for it to be added to https://github.com/kbuzard-SU/gradualism/

**Reply by ouuan** [reply]
> based on usage and customer feedback, we very well may add the ability to fully convert a discussion into an issue :)

Where can we track the progress of "the ability to fully convert a discussion into an issue"? Since this "question" already has an answer, this thread seems like already solved. Should we open a new thread for "fully convert"?

**Reply by native-api** [reply]
Need this in https://github.com/pyenv/pyenv for https://github.com/pyenv/pyenv/discussions/1974.

**Reply by ryandemopoulos** [reply]
Thanks for adding this capability @evi-liu!  I tried using it just now and it worked, but it would be very helpful if there were an option for the new Issue to be filed under the original filer's account.  I didn't see an option for this.

Example Use Case: if a user files a Discussion reporting a bug (that should have been an issue), I want to be able to convert the Discussion into an Issue but have that new Issue still list the original user as the one who opened it.

**Reply by nbouvrette** [reply]
https://github.com/Avansai/next-multilingual thanks

**Reply by gordsport** [reply]
Hey @evi-liu - how can we get access to `Create issue form this discussion` - this is something we would like in https://github.com/vegaprotocol/feedback/discussions

**Reply by artf** [reply]
@evi-liu is it possible to disable this feature? I have an issue form template enabled with `blank_issues_enabled: false` in the `config.yml` as I want to keep issues on my repo properly defined/described and this feature breaks the flow.

**Reply by timbomckay** [reply]
Has this feature been fully enabled or is it still in private beta? I see it in [the documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue#creating-an-issue-from-discussion) but our enterprise instance isn't showing this in our discussions.

**Comment by ouuan** [comment]
"Create issue from discussion" is useful for converting a feature request that was originally a Discussion to an Issue, but is not very useful if an Issue is mistakenly converted to a Discussion.

Maybe the currently converted Issues have already lost some information thus it's impossible to convert them back. What I would like is that in the future, when issues are converted to discussions, the old information can be preserved and the converting operation would be revertable.

To me, it's acceptable even if converting back is only available when there's no modification made after converting the issue to a discussion.

**Reply by wonderflow** [reply]
+1 after experience the feature, I found it is `create issue from discussion` not `convert discussion into issue`. \cc @evi-liu

**Reply by helmasaur** [reply]
What to important to he issue created in this case? Just the title with a link to the discussion? How to answer to something posed before the issue got created?

Or maybe you meant to create an issue with everything (like a clone) and locking the discussion?

I will be testing this functionnality tonight and tomorrow.

**Comment by bagder** [comment]
Note: **this** very repository has no 'issues' enabled and yet it allows us to create an issue out of a discussion? Maybe it shouldn't...

Snap:

![Screenshot_2021-05-07 Convert Discussion into Issue · Discussion #2861 · github feedback](https://user-images.githubusercontent.com/177011/117457017-200b0a80-af49-11eb-81a4-311ef9f7d8e3.png)

**Reply by helmasaur** [reply]
Have you tried clicking on it to know what happens?

**Reply by bagder** [reply]
No... I don't want to risk any damage or chaos.

**Reply by evi-liu** [reply]
Thanks for reporting this bug! We're tracking it on our end and will get a fix out for it soon.

**Comment by wiredsister** [comment]
I just wanted to comment that I really liked the feature but this difference between issue & discussion was confusing for me. I for one, did not realize you could make existing issues into discussions and now have a duplication issue 😬

**Reply by helmasaur** [reply]
I am not sure I understood what you mean. Is it: issue to disscussion and then back to issue that would create a second issue instead of using the first one?

**Reply by wiredsister** [reply]
I had issues and I didn't know they could become discussions so I made a bunch of duplicated discussions by accident. That's what I mean.

**Reply by adityaruplaha** [reply]
Alternatively, I would appreciate the ability to close/archive discussions.

**Reply by ouuan** [reply]
> Alternatively, I would appreciate the ability to close/archive discussions.

That's a different topic. See #3097.

**Comment by ivanperez-keera** [comment]
I also would like to be able to turn a discussion into an issue, in the same way that you can turn an issue into a discussion.

Is there a reason why it's not possible?

Creating an issue linked to a discussion is not the same, and it also requires additional clicks to gather context (now you have to go to the discussion and read it). Converting the discussion brings all context with it.

@evi-liu Can Github please implement the conversion?

**Comment by linas** [comment]
At this time, clicking on "create issue from discussion" just creates a brand-new issue, copies the first message only, and none of the rest of the thread. Worse, it lists me as the person creating the issue, and not the original reporter. In short, I lose track of everything: the issue reporter, the discussion, the list of people who will be notified on updates.  Is this really the intended operation, or have I misconfigured something?

(I mean, I'm always free to create new issues by hand, and link to the discussion. That's easy, and doesn't need a menu entry in the GUI (which just clogs up the GUI with a pointless menu entry). The hard part is to copy the list of subscribers over, and to copy the important parts of the discussion.

**Comment by axu2** [comment]
I created a discussion from an issue, which locked the original issue. I actually prefer the original issue format, is there any way to unlock the original issue? I get an error when I try to unlock.

- https://github.com/orgs/community/discussions/2952

EDIT: delete the converted discussion lets you unlock the issue.

**Comment by phifuh** [comment]
Is it possible to decide where the issue should be created? We have a central discussion hub and use labels to highlight the repo but I cant find the create issue button and neither any info about if this is even possible to select a specific repo in the orga
